### DISCUSSION TITLE
proof find vs repo query

### DISCUSSION THREAD

**Discussion by kornelski** [crev-dev/cargo-crev#452] [General]
It seems that these commands are almost duplicate:

```sh
cargo crev proof find --crate zeroize
cargo crev repo query review zeroize
```

I suggest moving `proof find` to be `repo query all`. This way there will be one top-level command less.

**Comment by dpc** [comment]
I was planing to retire `repo query`. :confused:

**Comment by kornelski** [comment]
Oh, `cargo crev id query` is `<thing you're searching for> query` vs `cargo crev repo query` is `<location being searched> query`.

So renaming `cargo crev id query` and `cargo crev crate search` to use the same verb would make sense.

**Comment by dpc** [comment]
`search` is for looking stuff up on crates.io and then sorting by reviews etc.

It's messy... But `search` is quite handy BTW. when I'm trying to find a crate to use for something.

**Comment by kornelski** [comment]
I'm not saying it's not useful, but just that terminology of `crate search` and `proof find` is jarring. The distinction between query, search and find seems accidental.

**Comment by dpc** [comment]
Yeaaaah... I know. I am not very good at naming... :smile:  . But how would you name them?
One is looking stuff by name in crates.io , another is for finding proofs.

`cargo crev crates.io-search`? :D

**Comment by kornelski** [comment]
creates.io search finds crates. If you want to stick to `noun verb` scheme, then a more consistent naming could be something like:

```
cargo crev crate find # searches for crates on crates.io
cargo crev proof find # searches for proofs in local cache
cargo crev id find # searches for ids in local cache
```

**Comment by dpc** [comment]
It does make sense!
